src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,67 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+import React, { ChangeEvent } from 'react';
+import { TodoStatus } from '../../types/TodoStatus';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+type Props = {
+  query: string,
+  setQuery: (input: React.SetStateAction<string>) => void,
+  setTodoFilter: (status: React.SetStateAction<TodoStatus>) => void,
+};
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+export const TodoFilter: React.FC<Props> = ({
+  query,
+  setQuery,
+  setTodoFilter,
+}) => {
+  const changeQuery = (event: ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+  };
+
+  const clearQuery = () => setQuery('');
+
+  const changeFilter = (event: ChangeEvent<HTMLSelectElement>) => {
+    setTodoFilter(event.target.value as TodoStatus);
+  };
+
+  return (
+    <form className="field has-addons">
+      <p className="control">
+        <span className="select">
+          <select
+            data-cy="statusSelect"
+            onChange={changeFilter}
+          >
+            <option value="all">{TodoStatus.All}</option>
```

Comment:
Let's use your enum for values too
```suggestion
            <option value={TodoStatus.All}>{TodoStatus.All}</option>
```